error when trying to template a repo

Tried to use it and it flamed, not really sure why. I didn't intentionally have any go templated things I expected to replace, but there are many different types of vars in the repo so maybe that is a problem?

steps to reproduce

```shell
ggft get https://github.com/raboley/review-infrastructure ri
ggft new ri aks
```

error message.

``` shell
# output
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x14d2a0b]

goroutine 1 [running]:
text/template.errRecover(0xc00014f650)
	/usr/local/opt/go/libexec/src/text/template/exec.go:163 +0x1c2
panic(0x1657ba0, 0x1c352c0)
	/usr/local/opt/go/libexec/src/runtime/panic.go:969 +0x166
text/template.(*Template).execute(0x0, 0x17f0800, 0xc00017fa70, 0x1653440, 0xc00017ea50, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/text/template/exec.go:217 +0x19b
text/template.(*Template).Execute(...)
	/usr/local/opt/go/libexec/src/text/template/exec.go:203
github.com/mjslabs/ggft/pkg/tmpl.ScanTemplateForVars(0xc000028d20, 0x4c, 0xe, 0x2f, 0xffffffffffffffff, 0x0, 0x2f00002e001b3520)
	/Users/russellboley/.go/pkg/mod/github.com/mjslabs/ggft@v0.0.1/pkg/tmpl/tmpl.go:48 +0x13b
github.com/mjslabs/ggft/cmd.getVarsFromUser(0xc000028d20, 0x4c, 0x5, 0x0)
	/Users/russellboley/.go/pkg/mod/github.com/mjslabs/ggft@v0.0.1/cmd/new.go:109 +0x3f
github.com/mjslabs/ggft/cmd.processPath(0xc000028d20, 0x4c, 0xc00002c900, 0x26, 0x7ffeefbff894, 0x3, 0x107d640, 0xc000028d70)
	/Users/russellboley/.go/pkg/mod/github.com/mjslabs/ggft@v0.0.1/cmd/new.go:98 +0x1de
github.com/mjslabs/ggft/cmd.walkTemplateDirectory.func1(0xc000028d20, 0x4c, 0x1802140, 0xc0001b3450, 0x0, 0x0, 0x13, 0xc00014f8d0)
	/Users/russellboley/.go/pkg/mod/github.com/mjslabs/ggft@v0.0.1/cmd/new.go:73 +0x59
path/filepath.walk(0xc000028d20, 0x4c, 0x1802140, 0xc0001b3450, 0xc00014fc30, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/path/filepath/path.go:360 +0x425
path/filepath.walk(0xc0000252c0, 0x38, 0x1802140, 0xc0001b32b0, 0xc00014fc30, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/path/filepath/path.go:384 +0x2ff
path/filepath.walk(0xc00002d050, 0x2e, 0x1802140, 0xc0001a9520, 0xc00014fc30, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/path/filepath/path.go:384 +0x2ff
path/filepath.walk(0xc00002c900, 0x26, 0x1802140, 0xc00013d5f0, 0xc00014fc30, 0x0, 0xc00002cc00)
	/usr/local/opt/go/libexec/src/path/filepath/path.go:384 +0x2ff
path/filepath.Walk(0xc00002c900, 0x26, 0xc00014fc30, 0x80000c00002cc00, 0x28)
	/usr/local/opt/go/libexec/src/path/filepath/path.go:406 +0xff
github.com/mjslabs/ggft/cmd.walkTemplateDirectory(0xc00002c900, 0x26, 0x7ffeefbff894, 0x3, 0xc00019a098, 0xc00017ec60)
	/Users/russellboley/.go/pkg/mod/github.com/mjslabs/ggft@v0.0.1/cmd/new.go:72 +0x83
github.com/mjslabs/ggft/cmd.newProject(0xc00002c900, 0x26, 0x7ffeefbff894, 0x3, 0x26, 0x0)
	/Users/russellboley/.go/pkg/mod/github.com/mjslabs/ggft@v0.0.1/cmd/new.go:68 +0x1c4
github.com/mjslabs/ggft/cmd.cmdNew(0x1c3e1a0, 0xc00000fe00, 0x2, 0x2)
	/Users/russellboley/.go/pkg/mod/github.com/mjslabs/ggft@v0.0.1/cmd/new.go:44 +0xeb
github.com/spf13/cobra.(*Command).execute(0x1c3e1a0, 0xc00000fdc0, 0x2, 0x2, 0x1c3e1a0, 0xc00000fdc0)
	/Users/russellboley/.go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0x1c3e440, 0x104380a, 0x1c01a00, 0xc000000180)
	/Users/russellboley/.go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/russellboley/.go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/mjslabs/ggft/cmd.Execute()
	/Users/russellboley/.go/pkg/mod/github.com/mjslabs/ggft@v0.0.1/cmd/root.go:40 +0x31
main.main()
	/Users/russellboley/.go/pkg/mod/github.com/mjslabs/ggft@v0.0.1/main.go:6 +0x20

```

--- a/cmd/new.go
+++ b/cmd/new.go
@@ -95,7 +95,9 @@ func processPath(curPath, templateDir, output string) error {
 		return nil
 	case mode.IsRegular():
 		// Template file, process accordingly
-		getVarsFromUser(curPath)
+		if err := getVarsFromUser(curPath); err != nil {
+			return err
+		}
 		return tmpl.CreateFileFromTemplate(curPath, destination)
 	}
 

--- a/pkg/tmpl/tmpl.go
+++ b/pkg/tmpl/tmpl.go
@@ -2,6 +2,7 @@ package tmpl
 
 import (
 	"bytes"
+	"errors"
 	"io/ioutil"
 	"os"
 	"regexp"
@@ -37,6 +38,9 @@ func CreateFileFromTemplate(input, output string) error {
 func ScanTemplateForVars(templateFile string) ([]string, error) {
 	buf := &bytes.Buffer{}
 	t, err := getTemplateObject(templateFile)
+	if err != nil {
+		return nil, errors.New(err.Error() + " in file " + templateFile)
+	}
 	r := regexp.MustCompile(`map has no entry for key "([^ ]+?)"`)
 	varList := []string{}
 

--- a/pkg/tmpl/tmpl_test.go
+++ b/pkg/tmpl/tmpl_test.go
@@ -12,6 +12,7 @@ import (
 func TestPkgTmpl(t *testing.T) {
 	t.Run("CreateFileFromTemplate", testCreateFile)
 	t.Run("ScanTemplateForVars", testScanTemplateForVars)
+	t.Run("ScanTemplateForVarsInvalid", testScanTemplateForVarsInvalid)
 }
 
 func testCreateFile(t *testing.T) {
@@ -36,5 +37,14 @@ func testScanTemplateForVars(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Len(t, varList, 1)
 	assert.Equal(t, varList[0], varName)
+	assert.NoError(t, os.Remove(tmplFile))
+}
+
+func testScanTemplateForVarsInvalid(t *testing.T) {
+	tmplFile := ".testing.tmpl"
+	assert.NoError(t, testhelpers.CreateFileWithContents(tmplFile, "{{ doesntexist.ForSure }} - {{ .TemplateFileName }}"))
+	_, err := ScanTemplateForVars(tmplFile)
+
+	assert.Error(t, err)
 	assert.NoError(t, os.Remove(tmplFile))
 }

